### PR TITLE
Create LogInServlet.java

### PR DESCRIPTION
**Pull request by cassie1992**

### CODE REVIEW

File: src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	@Override
+	public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException
+	{
+		DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+		Entity user = new Entity("User");
+		user.setProperty("username", "admin");
+		user.setProperty("password", "password");
+		user.setProperty("access", "Admin");
+		ds.put(user);
```
Comment (Ajedi32) [inline src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java:36]: So you create a new user every time someone tries to log in? :confused:

File: src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		Query q = new Query("User").setFilter(
+				Query.CompositeFilterOperator.and(
+						new Query.FilterPredicate("username",Query.FilterOperator.EQUAL,req.getParameter("username")),
+						new Query.FilterPredicate("password",Query.FilterOperator.EQUAL,req.getParameter("password"))
+						)
+					);
```
Comment (Ajedi32) [inline src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java:42]: Looks like the indentation here is a bit off.

File: src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		
+		if (ds.prepare(q).countEntities(FetchOptions.Builder.withDefaults()) > 0)
+		{
+			resp.sendRedirect("/homepage");
```
Comment (Ajedi32) [inline src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java:46]: Hmm... okay. I guess this is "the simplest thing that could possibly work". Before we start enforcing the log in requirement though the application will need some way of knowing whether or not the user is logged in. Are you planning to switch to using cookies after we start on that feature?

**Comment by Ajedi32** [comment]
Did you forget to change the web.xml file? The login page won't be accessible unless you add the appropriate routes there.

**Comment by Ajedi32** [comment]
Looks pretty good overall. Take a look at the comments I made above. If you want to change the pull request before I merge it, Cassie will have to push those changes to the `patch-1` branch on her repository.

File: src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java (hunk 1 above)
Comment (cassie1992) [inline src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java:36]: Oh yeah, I was going to move that somewhere else once/fix it up I just got it working, but I totally forgot about it. Sorry about that, I can fix that tonight

Cassie

----- Original Message -----

**Comment by cassie1992** [comment]
I was having issues adding everything (it might've been because I was tired), there should be a separate xml file (and the start of a homepage) in a different pull request. I can edit the web.xml page to include what's in my xml file

Cassie

----- Original Message -----

File: src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java (hunk 1 above)
Comment (Ajedi32) [inline src/ed/uwm/cs361/fantastic_five/training_tracker/LogInServlet.java:36]: No problem. :smile:

**Comment by Ajedi32** [comment]
@cassie1992 Sounds good. Keep up the good work. :+1:

**Comment by Ajedi32** [comment]
Closing in favor of #13.

**Pull request closed without merging**